src/sas/qtgui/UnitTesting/TestUtils.py

```diff
@@ -9,11 +9,11 @@ def WarningTestNotImplemented(method_name=None):
     Test name retrieved from stack trace.
     """
     if method_name is not None:
-        print(("\nWARNING: %s needs implementing!"%method_name))
+        print("\nWARNING: %s needs implementing!"%method_name)
```

Comment:
Is there a reason `%s` formatters were left when `"".format()` formatters were converted to f-strings?

Reply:
The format statements could be converted automatically, but there is not an automatic fix available for the `%s` formatters, and there are 504 instances of this in the code. I think this is something we will have to tackle more gradually. Note that this error is rule `UP031`, which I added to the list of rules to ignore.

Reply:
The fix safety is discussed here: https://docs.astral.sh/ruff/rules/printf-string-formatting/